craft_application/application.py

```diff
@@ -402,8 +402,10 @@ def run_managed(self, platform: str | None, build_for: str | None) -> None:
                 clean_existing=self._enable_fetch_service,
             ) as instance:
```

Comment:
Out-of-scope, but [this](https://github.com/canonical/craft-application/blob/654c1d5b9af1bd5b49fff26a0feafa0f0a7b947f/craft_application/application.py#L387-L427) and [this](https://github.com/canonical/craft-application/blob/654c1d5b9af1bd5b49fff26a0feafa0f0a7b947f/craft_application/services/provider.py#L422-L457) smell like duplicated code.

Reply:
You're right - it was an oversight for the `Application` class to still have a `run_managed`